lib/rubygems/package/tar_header.rb

```diff
@@ -172,6 +172,15 @@ def initialize(vals)
     @empty = vals[:empty]
   end
 
+  EMPTY = new({ # :nodoc:
+    mode: 0,
+    name: "",
+    prefix: "",
+    size: 0,
+    empty: true,
+  }).freeze
```

Comment:
To really be identical you need the following. The default values are not the same as what an empty header outputs.

```suggestion
  EMPTY = new({ # :nodoc:
    checksum: 0,
    gname: "",
    linkname: "",
    magic: "",
    mode: 0,
    name: "",
    prefix: "",
    size: 0,
    uname: "",
    version: 0,
    empty: true,
  }).freeze
```
Here's my irb doublecheck:

```ruby
irb(main):033> EMPTY = Gem::Package::TarHeader.new({ checksum: 0, gname: "", linkname: "", magic: "", mode: 0, name: "", prefix: "", size: 0, uname: "", version: 0, empty: true,  })
=> #<Gem::Package::TarHeader:0x00000001203e60e0 @checksum=0, @devmajor=0, @devminor=0, @empty=true, @gid=0, @gname="", @linkname="", @magic="", @mode=0, @mtime=0, @name="", @prefix="", @size=0, @typeflag="0", @uid=0, @uname="", @version=0>
irb(main):034> Gem::Package::TarHeader.from(StringIO.new(Gem::Package::TarHeader::EMPTY_HEADER))
=> #<Gem::Package::TarHeader:0x00000001204cde90 @checksum=0, @devmajor=0, @devminor=0, @empty=true, @gid=0, @gname="", @linkname="", @magic="", @mode=0, @mtime=0, @name="", @prefix="", @size=0, @typeflag="0", @uid=0, @uname="", @version=0>
```


We should also add a test.